### ISSUE TITLE
annotate_variants_expr breaks genotype columns with bgen data

### ISSUE THREAD

**Issue by jkeebler** [hail-is/hail#1960]
```
> sf_vds.genotypes_table().to_dataframe().select(['`g.gt`','`g.px`','`g.isLinearScale`']).show(5)
+----+-------------+----------------+
|g.gt|         g.px| g.isLinearScale|
+----+-------------+----------------+
|   0|[32768, 0, 0]|            True|
|   0|[32768, 0, 0]|            True|
|   0|[32768, 0, 0]|            True|
|   0|[32768, 0, 0]|            True|
|   0|[32768, 0, 0]|            True|
+----+-------------+----------------+
only showing top 5 rows

> sf_va_vds = sf_vds.annotate_variants_expr('va.callRate = gs.fraction(g => g.isCalled())')

> sf_va_vds.genotypes_table().to_dataframe().select(['`g.gt`','`g.px`','`g.isLinearScale`']).show(5)
+----+----+----------------+
|g.gt|g.px| g.isLinearScale|
+----+----+----------------+
|null|null|            True|
|null|null|            True|
|null|null|            True|
|null|null|            True|
|null|null|            True|
+----+----+----------------+
only showing top 5 rows
```

**Comment by tpoterba** [comment]
Addressed by #1956

**Comment by tpoterba** [comment]
Fixed
